### ISSUE TITLE
run error (with nerfstudio 0.3.4)

### ISSUE THREAD

**Issue by qhdqhd** [hturki/pynerf#10]
```
`(nerfstudio) qhd@qhd-H610M4:~/code/pynerf-main$ ns-train pynerf --data /home/qhd/code/data/mine/Replica/re_scan2 nerfstudio-data

optimizer is no longer specified in the CameraOptimizerConfig, it is now defined with the rest of the param groups 
inside the config file under the name 'camera_opt'

/home/qhd/code/pynerf-main/pynerf/data/datamanagers/random_subset_datamanager.py:42: FutureWarning: above message coming from
  camera_optimizer: CameraOptimizerConfig = CameraOptimizerConfig(

scheduler is no longer specified in the CameraOptimizerConfig, it is now defined with the rest of the param groups 
inside the config file under the name 'camera_opt'


optimizer is no longer specified in the CameraOptimizerConfig, it is now defined with the rest of the param groups 
inside the config file under the name 'camera_opt'

/home/qhd/anaconda3/envs/nerfstudio/lib/python3.8/site-packages/tyro/_calling.py:245: FutureWarning: above message coming from
  return unwrapped_f(*positional_args, **kwargs), consumed_keywords  # type: ignore

scheduler is no longer specified in the CameraOptimizerConfig, it is now defined with the rest of the param groups 
inside the config file under the name 'camera_opt'

[13:58:39] Using --data alias for --data.pipeline.datamanager.data                                          train.py:230
──────────────────────────────────────────────────────── Config ────────────────────────────────────────────────────────
TrainerConfig(
    _target=<class 'nerfstudio.engine.trainer.Trainer'>,
    output_dir=PosixPath('outputs'),
    method_name='pynerf',
    experiment_name=None,
    project_name='nerfstudio-project',
    timestamp='2024-08-05_135839',
    machine=MachineConfig(seed=42, num_devices=1, num_machines=1, machine_rank=0, dist_url='auto', device_type='cuda'),
    logging=LoggingConfig(
        relative_log_dir=PosixPath('.'),
        steps_per_log=10,
        max_buffer_size=20,
        local_writer=LocalWriterConfig(
            _target=<class 'nerfstudio.utils.writer.LocalWriter'>,
            enable=True,
            stats_to_track=(
                <EventName.ITER_TRAIN_TIME: 'Train Iter (time)'>,
                <EventName.TRAIN_RAYS_PER_SEC: 'Train Rays / Sec'>,
                <EventName.CURR_TEST_PSNR: 'Test PSNR'>,
                <EventName.VIS_RAYS_PER_SEC: 'Vis Rays / Sec'>,
                <EventName.TEST_RAYS_PER_SEC: 'Test Rays / Sec'>,
                <EventName.ETA: 'ETA (time)'>
            ),
            max_log_size=10
        ),
        profiler='basic'
    ),
    viewer=ViewerConfig(
        relative_log_filename='viewer_log_filename.txt',
        websocket_port=None,
        websocket_port_default=7007,
        websocket_host='0.0.0.0',
        num_rays_per_chunk=32768,
        max_num_display_images=512,
        quit_on_train_completion=False,
        image_format='jpeg',
        jpeg_quality=75,
        make_share_url=False,
        camera_frustum_scale=0.1,
        default_composite_depth=True
    ),
    pipeline=PyNeRFPipelineConfig(
        _target=<class 'pynerf.data.pynerf_pipelines.PyNeRFPipeline'>,
        datamanager=RandomSubsetDataManagerConfig(
            _target=<class 'pynerf.data.datamanagers.random_subset_datamanager.RandomSubsetDataManager'>,
            data=PosixPath('/home/qhd/code/data/mine/Replica/re_scan2'),
            masks_on_gpu=False,
            images_on_gpu=False,
            dataparser=NerfstudioDataParserConfig(
                _target=<class 'nerfstudio.data.dataparsers.nerfstudio_dataparser.Nerfstudio'>,
                data=PosixPath('.'),
                scale_factor=1.0,
                downscale_factor=None,
                scene_scale=3.0,
                orientation_method='up',
                center_method='poses',
                auto_scale_poses=True,
                eval_mode='fraction',
                train_split_fraction=0.9,
                eval_interval=8,
                depth_unit_scale_factor=0.001
            ),
            train_num_rays_per_batch=8192,
            eval_num_rays_per_batch=4096,
            eval_image_indices=None,
            camera_optimizer=CameraOptimizerConfig(
                _target=<class 'nerfstudio.cameras.camera_optimizers.CameraOptimizer'>,
                mode='off',
                trans_l2_penalty=0.01,
                rot_l2_penalty=0.001,
                optimizer=AdamOptimizerConfig(
                    _target=<class 'torch.optim.adam.Adam'>,
                    lr=6e-06,
                    eps=1e-15,
                    max_norm=None,
                    weight_decay=0
                ),
                scheduler=ExponentialDecaySchedulerConfig(
                    _target=<class 'nerfstudio.engine.schedulers.ExponentialDecayScheduler'>,
                    lr_pre_warmup=1e-08,
                    lr_final=0.0006,
                    warmup_steps=0,
                    max_steps=125000,
                    ramp='cosine'
                )
            ),
            items_per_chunk=25600000,
            local_cache_path='scratch/pynerf-cache',
            on_demand_threads=16,
            load_all_in_memory=False
        ),
        model=PyNeRFModelConfig(
            _target=<class 'pynerf.models.pynerf_model.PyNeRFModel'>,
            enable_collider=True,
            collider_params={'near_plane': 0.0, 'far_plane': 20.0},
            loss_coefficients={'rgb_loss_coarse': 1.0, 'rgb_loss_fine': 1.0},
            eval_num_rays_per_chunk=4096,
            prompt=None,
            near_plane=0.2,
            far_plane=1000.0,
            geo_feat_dim=15,
            num_layers=2,
            num_layers_color=3,
            hidden_dim=64,
            hidden_dim_color=128,
            num_levels=16,
            base_resolution=16,
            max_resolution=2048,
            log2_hashmap_size=20,
            features_per_level=4,
            appearance_embedding_dim=32,
            disable_scene_contraction=False,
            background_color='last_sample',
            output_interpolation='embedding',
            level_interpolation='linear',
            num_scales=8,
            scale_factor=2,
            share_feature_grid=True,
            num_proposal_samples_per_ray=(256, 96),
            num_nerf_samples_per_ray=48,
            proposal_update_every=5,
            proposal_warmup=5000,
            num_proposal_iterations=2,
            use_same_proposal_network=False,
            interlevel_loss_mult=1.0,
            distortion_loss_mult=0.002,
            use_proposal_weight_anneal=True,
            proposal_weights_anneal_slope=10.0,
            proposal_weights_anneal_max_num_iters=1000,
            use_single_jitter=True,
            use_gradient_scaling=False
        )
    ),
    optimizers={
        'mlps': {
            'optimizer': AdamOptimizerConfig(
                _target=<class 'torch.optim.adam.Adam'>,
                lr=0.01,
                eps=1e-15,
                max_norm=None,
                weight_decay=1e-09
            ),
            'scheduler': CosineDecaySchedulerConfig(
                _target=<class 'nerfstudio.engine.schedulers.CosineDecayScheduler'>,
                warm_up_end=512,
                learning_rate_alpha=0.05,
                max_steps=30000
            )
        },
        'fields': {
            'optimizer': AdamOptimizerConfig(
                _target=<class 'torch.optim.adam.Adam'>,
                lr=0.01,
                eps=1e-15,
                max_norm=None,
                weight_decay=1e-09
            ),
            'scheduler': CosineDecaySchedulerConfig(
                _target=<class 'nerfstudio.engine.schedulers.CosineDecayScheduler'>,
                warm_up_end=512,
                learning_rate_alpha=0.05,
                max_steps=30000
            )
        },
        'proposal_networks': {
            'optimizer': AdamOptimizerConfig(
                _target=<class 'torch.optim.adam.Adam'>,
                lr=0.01,
                eps=1e-15,
                max_norm=None,
                weight_decay=1e-09
            ),
            'scheduler': CosineDecaySchedulerConfig(
                _target=<class 'nerfstudio.engine.schedulers.CosineDecayScheduler'>,
                warm_up_end=512,
                learning_rate_alpha=0.05,
                max_steps=30000
            )
        }
    },
    vis='viewer',
    data=PosixPath('/home/qhd/code/data/mine/Replica/re_scan2'),
    prompt=None,
    relative_model_dir=PosixPath('nerfstudio_models'),
    load_scheduler=True,
    steps_per_save=1000,
    steps_per_eval_batch=500,
    steps_per_eval_image=5000,
    steps_per_eval_all_images=30000,
    max_num_iterations=30001,
    mixed_precision=True,
    use_grad_scaler=False,
    save_only_latest_checkpoint=True,
    load_dir=None,
    load_step=None,
    load_config=None,
    load_checkpoint=None,
    log_gradients=False,
    gradient_accumulation_steps={}
)
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
           Saving config to: outputs/re_scan2/pynerf/2024-08-05_135839/config.yml               experiment_config.py:141
           Saving checkpoints to: outputs/re_scan2/pynerf/2024-08-05_135839/nerfstudio_models             trainer.py:134
           Auto image downscale factor of 1                                                 nerfstudio_dataparser.py:389
Traceback (most recent call last):
  File "/home/qhd/anaconda3/envs/nerfstudio/bin/ns-train", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/qhd/code/nerfstudio/nerfstudio/scripts/train.py", line 262, in entrypoint
    main(
  File "/home/qhd/code/nerfstudio/nerfstudio/scripts/train.py", line 247, in main
    launch(
  File "/home/qhd/code/nerfstudio/nerfstudio/scripts/train.py", line 189, in launch
    main_func(local_rank=0, world_size=world_size, config=config)
  File "/home/qhd/code/nerfstudio/nerfstudio/scripts/train.py", line 99, in train_loop
    trainer.setup()
  File "/home/qhd/code/nerfstudio/nerfstudio/engine/trainer.py", line 147, in setup
    self.pipeline = self.config.pipeline.setup(
  File "/home/qhd/code/nerfstudio/nerfstudio/configs/base_config.py", line 54, in setup
    return self._target(self, **kwargs)
  File "/home/qhd/code/nerfstudio/nerfstudio/pipelines/base_pipeline.py", line 258, in __init__
    self.datamanager: DataManager = config.datamanager.setup(
  File "/home/qhd/code/nerfstudio/nerfstudio/configs/base_config.py", line 54, in setup
    return self._target(self, **kwargs)
  File "/home/qhd/code/pynerf-main/pynerf/data/datamanagers/random_subset_datamanager.py", line 96, in __init__
    self.train_ray_generator = RayGenerator(self.train_dataparser_outputs.cameras.to(self.device),
TypeError: __init__() takes 2 positional arguments but 3 were given
```

**Comment by hturki** [comment]
See https://github.com/hturki/pynerf/issues/6

**Comment by qhdqhd** [comment]
thank a lot!
It successfully training by change this:
```
+++ b/pynerf/data/datamanagers/random_subset_datamanager.py
@@ -93,8 +93,7 @@ class RandomSubsetDataManager(DataManager):

         self.train_camera_optimizer = self.config.camera_optimizer.setup(
             num_cameras=self.train_dataparser_outputs.cameras.size, device=self.device)
-        self.train_ray_generator = RayGenerator(self.train_dataparser_outputs.cameras.to(self.device),
-                                                self.train_camera_optimizer)
+        self.train_ray_generator = RayGenerator(self.train_dataparser_outputs.cameras.to(self.device))

         fields_to_load = {RGB}
         for additional_field in {DEPTH, WEIGHT, TRAIN_INDEX}:
@@ -117,8 +116,7 @@ class RandomSubsetDataManager(DataManager):
         self.eval_dataset = InputDataset(self.eval_dataparser_outputs)
         self.eval_camera_optimizer = self.config.camera_optimizer.setup(
             num_cameras=self.eval_dataparser_outputs.cameras.size, device=self.device)
-        self.eval_ray_generator = RayGenerator(self.eval_dataparser_outputs.cameras.to(self.device),
-                                               self.eval_camera_optimizer)
+        self.eval_ray_generator = RayGenerator(self.eval_dataparser_outputs.cameras.to(self.device))

         self.eval_image_metadata = self._get_image_metadata(self.eval_dataparser_outputs)
         self.eval_batch_dataset = RandomSubsetDataset(
```

However, some error happened in eval:
```
  File "/home/qhd/code/nerfstudio/nerfstudio/pipelines/base_pipeline.py", line 342, in get_eval_image_metrics_and_images
    camera, batch = self.datamanager.next_eval_image(step)
ValueError: too many values to unpack (expected 2)
```

**Comment by qhdqhd** [comment]
solved! I use nerfstudio v0.3.3.
